src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs

```diff
@@ -18,10 +18,15 @@ public sealed class StorageAccountAttribute : Attribute
         /// <summary>
         /// Constructs a new instance.
         /// </summary>
-        /// <param name="account">The name of the Azure Storage connection string to use. This should be the name
-        /// of one of the connection strings present in the application settings (or environment variables). The
-        /// connection string name in settings should be prefixed with the standard "AzureWebJobs" prefix, but the
-        /// value you specify here should not include that prefix.</param>
+        /// <param name="account">A string value indicating the Azure Storage connection string to use. This
+        /// string should be in one of the following formats. These checks will be applied in order and the
+        /// first match wins.
+        /// - The name of an "AzureWebJobs" prefixed app setting or connection string name. E.g., if your setting
+        ///   name is "AzureWebJobsMyStorage", you can specify "MyStorage" here.
+        /// - Can be a string containing %% values (e.g. %StagingStorage%). The value provided will be passed
+        ///   to any INameResolver registered on the JobHostConfiguration to resolve the actual setting name to use.
+        /// - Can be an app setting or connection string name of your choosing.
+        /// </param>
         public StorageAccountAttribute(string account)
```

Comment:
I didn't realize this existed until now. It seems generally useful. Can this be named something like ConnectionNameAttribute and be used by other bindings? I'm thinking we need to add this behavior throughout the extensions repo.

If it's too hard to generalize this one (I haven't looked through it all yet), maybe we could add it there and allow all those extensions to re-use it?